ocaml/xapi/cert_distrib.ml

```diff
@@ -399,29 +399,83 @@ let exchange_certificates_among_all_members ~__context =
     (fun host -> Worker.remote_regen_bundle host rpc session_id)
     all_hosts
 
-let import_joiner ~__context ~uuid ~certificate ~to_hosts =
-  let joiner_certificate =
-    HostPoolProvider.certificate_of_id_content uuid certificate
+let ( (get_local_ca_certs : unit -> WireProtocol.certificate_file list)
+    , (get_local_pool_certs : unit -> WireProtocol.certificate_file list) ) =
+  let g path () =
+    Sys.readdir path
+    |> Array.to_list
+    |> List.map (fun filename ->
```

Comment:
Should we filter for `Filename.check_suffix` to only include `*.pem` but no `*.bak` or similar?

Reply:
I think you are right and we should take only `*.pem` (since `update-ca-bundle.sh` generates the bundle only using `*.pem` files, so anything else does not end up in the bundle)